packages/views.client-react.unstable/src/client.tsx

```diff
@@ -16,15 +16,17 @@
 
 import type {
   AsyncParameterValueMap,
-  ParameterConfig,
+  ViewConfig,
 } from "@osdk/views-client.unstable";
 import { createFoundryViewClient } from "@osdk/views-client.unstable";
 import React, { useEffect, useMemo } from "react";
 import type { FoundryViewClientContext } from "./context.js";
 import { FoundryViewContext } from "./context.js";
 import { initializeParameters } from "./utils/initializeParameters.js";
 
-interface FoundryViewProps<CONFIG extends ParameterConfig> {
+interface FoundryViewProps<
+  CONFIG extends ViewConfig<{ parameters: CONFIG["parameters"] }>,
```

Comment:
Were you not getting the types you wanted without this?

Reply:
correct, it would not narrow in the event type and instead make every event's parameters be a union of all parameters defined in the event config :(